pilot/pkg/xds/delta.go

```diff
@@ -475,7 +468,8 @@ func (s *DiscoveryServer) pushDeltaXds(con *Connection, push *model.PushContext,
 	if len(resp.RemovedResources) > 0 {
 		log.Debugf("ADS:%v %s REMOVE %v", v3.GetShortType(w.TypeUrl), con.ConID, resp.RemovedResources)
 	}
-	if isWildcardTypeURL(w.TypeUrl) {
+	// normally wildcard xds `subscribe` is always nil, just in case there are some extended type not handled correctly.
+	if subscribe == nil && isWildcardTypeURL(w.TypeUrl) {
 		// this is probably a bad idea...
```

Comment:
can wildcard ever have non nil subscribe? That seems like it should be impossible..?

Reply:
Generally yes, just in case a wrong xds client implement, especially our extended type